puppet/lib/puppet/util/puppetdb/config.rb

```diff
@@ -14,6 +14,8 @@ def self.load(config_file = nil)
         :soft_write_failure        => false,
```

Comment:
I think soft_write_failure is the same as min_successful_writes=0, isn't it? I'll just make them use the same mechanism and forbid configuring them together.